threadpoolctl.py

```diff
@@ -383,6 +397,36 @@ def _get_current_backend(self):
         get_backend_(backend, ctypes.sizeof(backend))
         return backend.value.decode("utf-8")
 
+    def switch_backend(self, backend):
+        """Switch the backend of FlexiBLAS
+
+        Parameters
+        ----------
+        backend : str
+            The name or the path to the shared library of the backend to switch to. If
+            the backend is not already loaded, it will be loaded first.
+        """
+        if backend not in self.loaded_backends:
+            if backend in self.available_backends:
+                load_func = getattr(self.dynlib, "flexiblas_load_backend", lambda _: -1)
+            else:  # assume backend is a path to a shared library
+                load_func = getattr(
+                    self.dynlib, "flexiblas_load_backend_library", lambda _: -1
+                )
+            res = load_func(str(backend).encode("utf-8"))
+            if res == -1:
+                raise RuntimeError(f"Failed to load backend {backend!r}.")
```

Comment:
I added details in the error message but it's hard to specialize the error too much because the user gives a string and we don't know if it's meant to be a name or a path. I don't think it's worth adding too much complexity to try to infer that.